lib/polyfill/vttcue.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+goog.provide('shaka.polyfill.VTTCue');
+
+goog.require('shaka.log');
+goog.require('shaka.polyfill.register');
+
+
+/**
+ * @namespace shaka.polyfill.VTTCue
+ *
+ * @summary A polyfill to provide VTTCue.
+ */
+
+
+/**
+ * Install the polyfill if needed.
+ */
+shaka.polyfill.VTTCue.install = function() {
+  if (window.VTTCue) {
+    shaka.log.info('Using native VTTCue.');
+    return;
+  }
+
+  if (window.TextTrackCue &&
+      window.TextTrackCue.prototype.constructor.length == 7) {
+    shaka.log.info('Using VTTCue polyfill from 7 argument TextTrackCue.');
+    window.VTTCue = shaka.polyfill.VTTCue.from7ArgsTextTrackCue_;
+  } else {
+    shaka.log.info('VTTCue not available.');
+  }
+};
+
+
+/**
+ * Draft spec TextTrackCue with 7 constructor arguments.
+ * See {@link https://goo.gl/CSPvbz Using the TextTrack API} and examples.
```

Comment:
This link doesn't specify the 7 arguments or their meaning.  It gives an example with 4 arguments, and the examples have the text in argument 4, not argument 1 as you have below.

I tracked down the history as best I could.

Here we had the first appearance of TextTrackCue in [draft 20110113](https://www.w3.org/TR/2011/WD-html5-20110113/video.html#texttrackcue) (no constructor).

In [draft 20120329](https://www.w3.org/TR/2012/WD-html5-20120329/media-elements.html#texttrackcue), we had the first constructor on TextTrackCue (6 arguments, not 7).

In [draft 20121025](https://www.w3.org/TR/2012/WD-html5-20121025/media-elements.html#texttrackcue), the constructor was changed to 3 arguments (as implemented by IE11 and Edge today).

In [draft 20121217](https://www.w3.org/TR/2012/CR-html5-20121217/embedded-content-0.html#texttrackcue), we have the final appearance of a TextTrackCue constructor.  After this, it became abstract.

Are you sure it's 7 on Toshiba and not 6?